packages/core/src/accessibility/useInteractiveAttributes.ts

```diff
@@ -0,0 +1,88 @@
+/* !
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ */
+
+import * as React from "react";
+
+import { mergeRefs, Utils } from "../common";
+
+type InteractiveHTMLAttributes<E extends HTMLElement> = Pick<
+    React.HTMLAttributes<E>,
+    "onBlur" | "onClick" | "onFocus" | "onKeyDown" | "onKeyUp" | "tabIndex"
+>;
+
+interface InteractiveComponentProps extends InteractiveHTMLAttributes<HTMLElement> {
+    active?: boolean | undefined;
+}
+
+interface InteractiveAttributes<E extends HTMLElement> extends InteractiveHTMLAttributes<E> {
+    ref: React.Ref<E>;
+}
+
+export function useInteractiveAttributes<E extends HTMLElement>(
+    interactive: boolean,
+    props: InteractiveComponentProps,
+    ref: React.Ref<E>,
+    defaultTabIndex?: number,
+): [active: boolean, interactiveProps: InteractiveAttributes<E>] {
+    const { active, onClick, onFocus, onKeyDown, onKeyUp, onBlur, tabIndex = defaultTabIndex } = props;
+    // the current key being pressed
+    const [currentKeyPressed, setCurrentKeyPressed] = React.useState<string | undefined>();
+    // whether the button is in "active" state
+    const [isActive, setIsActive] = React.useState(false);
+    // our local ref for the interactive element, merged with the consumer's own ref (if supplied) in this hook's return value
```

Comment:
**nit:** since `ref` is not an optional argument within this hook, can we change the wording of this comment?
```suggestion
    // our local ref for the interactive element, merged with the consumer's own ref in this hook's return value
```